components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -109,6 +109,19 @@ void ViewCounterModel::RegisterPageViewForBackgroundImages() {
   current_wallpaper_image_index_ %= total_image_count_;
 }
 
+void ViewCounterModel::SetBrandedWallpaperRetrieveFailed() {
```

Comment:
nit: It would be good to add some comments why background wallpaper index is increasted.
IMO, `IncreaseBackgroundWallpaperImageIndex()` would be more appropriate because this model class doesn't know about this failure.
and adding comment about why count is increasted to the call site seems much better. WDYT?

Reply:
sounds good! I will change.